### DISCUSSION TITLE
createId(), messaging.messages and messaging.tokenChanges in v7

### DISCUSSION THREAD

**Discussion by metalllus** [angular/angularfire#3062] [Q&A]
Hi, I was wondering what are the equivalents to these methods in Angularfire v.7

1. Firestore - `createId()` method which should create an id for a document to be created in Firestore
2. Messaging - `tokenChanges()` method which should return an observable emitting the token every time there is a change
3. Messaging - `message()` method which should return an observable emitting the push notifications from FCM.

I am migrating to v7 from v6 and I was not able to find equivalents for the methods above. Any help finding the new equivalents would be greatly appreciated.

**Comment by johnnycopes** [comment]
From what I can tell, the `createId` function is no longer available. However, looking at [its source code](https://github.com/angular/angularfire/blob/master/src/compat/firestore/firestore.ts#L237-L239), it seems like a pretty simple utility function that you can define yourself like so:

```
import { collection, doc } from '@angular/fire/firestore';

createId(): string {
  return doc(collection(FIRESTORE_INSTANCE, '_')).id;
}
```
